src/main/java/org/fogbowcloud/app/core/monitor/JobSubmissionMonitor.java

```diff
@@ -27,29 +27,30 @@ public JobSubmissionMonitor(JobDataStore jobDataStore,
     @Override
     public void run() {
         LOGGER.debug("Checking for jobs to be submitted.");
-        JDFJob job = this.jobsBuffer.peek();
-        if (Objects.nonNull(job)) {
+        while (Objects.nonNull(this.jobsBuffer.peek())) {
+            JDFJob job = this.jobsBuffer.poll();
             LOGGER.debug("Job found! Starting job submission with id [" + job.getId() + "]");
             try {
                 final String arrebolId = this.jobExecutionSystem.execute(job);
                 job.setJobIdArrebol(arrebolId);
                 job.setState(JDFJobState.SUBMITTED);
                 this.jobDataStore.update(job);
-                this.jobsBuffer.poll();
+
                 LOGGER.info(
                     "Iguassu Job [" + job.getId() + "] has arrebol id: [" + job.getJobIdArrebol()
                         + "]");
 
             } catch (ArrebolConnectException ace) {
                 LOGGER.error("Job execution service is not available right now.");
+                break;
```

Comment:
Add the job back to the queue.

Reply:
Solved in 237525f99fb1ea9d1c59a80fda501accf8dd5a98.